### ISSUE TITLE
Orca: Hook Mechanism TaskBoard

### ISSUE THREAD

**Issue by leonardozcm** [intel-analytics/BigDL#7105]
## Following Tasks

- [ ] Insert Base Hook Func into TorchRunner https://github.com/intel-analytics/BigDL/pull/7100
- [ ] In `before_train_iter`, restore batch: [x1, x2], y -> x1, x2, y
- [ ] Assign some attrs to runner(self.batch, self.loss etc.)
- [ ] Implement MainCallBacks
- [ ] Migrate Existing CallBacks: TensorBoard, ModelCheckPoint
- [ ] CleanTorchRunner
- [ ] User Interfaces: estimator.fit
- [ ] Example: Resnet50 Inference
- [ ] Doc

**Comment by jenniew** [comment]
For huggingface based training:
- [ ] Support model input is a dictionary which includes labels. Also in current torch runner, if feature is not list nor tensor, it will trigger error. We may remove this restriction. Maybe put all these data processing and check to hook function and provide default implementation to support various data source type. 
- [ ] Support train with max steps
- [ ] Support ModelCheckpoint on step_frequency

**Comment by leonardozcm** [comment]
- [ ] Pre-check tqdmcallback
- [ ] Add tqdm supports in evaluate and predict
- [ ] Suggest users to write all logics in Maincallback.

**Comment by songhappy** [comment]
Orca estimator needs to support multiple outputs
https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py#L647
https://github.com/intel-analytics/BigDL/issues/7348

**Comment by leonardozcm** [comment]
> Orca estimator needs to support multiple outputs code: https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py#L647
> 
> issue: #7348

https://github.com/intel-analytics/BigDL/actions/runs/4219544101/jobs/7325009597

**Comment by leonardozcm** [comment]
https://github.com/intel-analytics/BigDL/issues/7433#issuecomment-1439073600
